modules/uolBidAdapter.js

```diff
@@ -0,0 +1,202 @@
+        isValid = false;
+      }
+      if (typeof bid.params.cpmFactor != 'undefined' && !bid.params.test) {
+        utils.logError(UOL_LOG_HEADER + 'Cannot manipulate cpmFactor outside test environment - bidID ' + bid.bidId);
+        isValid = false;
+      }
+      if (bid.params.cpmFactor && (isNaN(bid.params.cpmFactor) || parseInt(Number(bid.params.cpmFactor)) != bid.params.cpmFactor || isNaN(parseInt(bid.params.cpmFactor, 10)))) {
+        utils.logError(UOL_LOG_HEADER + 'Invalid param definition for cpmFactor on bidID ' + bid.bidId);
+        isValid = false;
+      }
+    } else {
+      isValid = false;
+      utils.logError(UOL_LOG_HEADER + 'No params defined for bidID ' + bid.bidId);
+    }
+    if (getSizes(bid).length == 0) {
+      utils.logError(UOL_LOG_HEADER + 'No sizing definition found for bidID ' + bid.bidId);
+      isValid = false;
+    }
+    return isValid;
+  },
+  /**
+    * Make a server request from the list of BidRequests.
+    *
+    * @param {validBidRequests[]} - an array of bids
+    * @param {bidderRequests} - an object containing all bid params, including validBids.
+    * @return ServerRequest Info describing the request to the server.
+    */
+  buildRequests: function(validBidRequests, bidderRequest) {
+    var data = JSON.stringify(getPayloadFor(validBidRequests, bidderRequest));
+    return {
+      method: 'POST',
+      url: ENDPOINT_URL,
+      data,
+      bidRequest: validBidRequests,
+    };
+  },
+  /**
+     * Unpack the response from the server into a list of bids.
+     *
+     * @param {ServerResponse} serverResponse A successful response from the server.
+     * @return {Bid[]} An array of bids which were nested inside the server.
+     */
+  interpretResponse: function(serverResponse, {bidRequest}) {
+    const bidResponses = [];
+    if (serverResponse.body) {
+      const ads = serverResponse.body.ads;
+      for (var index = 0; index < ads.length; index++) {
+        const bidResponse = {
+          requestId: ads[index].bidId,
+          cpm: ads[index].cpm,
+          width: ads[index].width,
+          height: ads[index].height,
+          creativeId: ads[index].creativeId,
+          dealId: ads[index].dealId,
+          currency: ads[index].currency,
+          netRevenue: ads[index].netRevenue,
+          ttl: ads[index].ttl,
+          ad: ads[index].ad,
+          mediaType: ads[index].mediaType
+        };
+        bidResponses.push(bidResponse);
+      }
+    }
+    return bidResponses;
+  },
+
+  /**
+     * Register the user sync pixels which should be dropped after the auction.
+     *
+     * @param {SyncOptions} syncOptions Which user syncs are allowed?
+     * @param {ServerResponse[]} serverResponses List of server's responses.
+     * @return {UserSync[]} The user syncs which should be dropped.
+     */
+  getUserSyncs: function(syncOptions, serverResponses) {
+    const syncs = [];
+    if (syncOptions.iframeEnabled && serverResponses.body.trackingPixel) {
```

Comment:
`serverResponses` is an array of objects, not a single object.  This latter half of the code would result in undefined errors.  Could you update this logic?